prov/efa/src/rxr/rxr_ep.c

```diff
@@ -1596,12 +1597,30 @@ void rxr_ep_progress_internal(struct rxr_ep *ep)
 	dlist_foreach_container_safe(&ep->tx_entry_queued_list,
 				     struct rxr_tx_entry,
 				     tx_entry, queued_entry, tmp) {
-		if (tx_entry->state == RXR_TX_QUEUED_CTRL)
-			ret = rxr_pkt_post_ctrl(ep, RXR_TX_ENTRY, tx_entry,
-						tx_entry->queued_ctrl.type,
-						tx_entry->queued_ctrl.inject);
-		else
+		if (tx_entry->state == RXR_TX_QUEUED_CTRL) {
+			peer = rxr_ep_get_peer(ep, tx_entry->addr);
+			if (tx_entry->queued_ctrl.type >= RXR_DC_REQ_PKT_BEGIN &&
+			    tx_entry->queued_ctrl.type <= RXR_DC_REQ_PKT_END) {
+				if (!(peer->flags & RXR_PEER_HANDSHAKE_RECEIVED))
+					ret = -FI_EAGAIN;
+				else if (!rxr_peer_support_delivery_complete(peer))
```

Comment:
here we use 

```peer->features[0] & REQ_INF_LIST[tx_entry->queued_ctrl.type].ex_feature_flag```

Reply:
queueing logic no longer used. resolving.